server/medical-contact/middleware.ts

```diff
@@ -0,0 +1,115 @@
+import type {Request, Response, NextFunction} from 'express';
+import {Types} from 'mongoose';
+import MedicalContactCollection from './collection';
+
+/**
+ * Checks if a medical contact with medicalContactId in req.params exists
+ */
+const isMedicalContactExists = async (req: Request, res: Response, next: NextFunction) => {
+    const validFormat = Types.ObjectId.isValid(req.params.medicalContactId);
+    const medicalContact = validFormat ? await MedicalContactCollection.findOne(req.params.medicalContactId) : '';
+    if (!medicalContact) {
+      res.status(404).json({
+        error: `Medical contact with medical contact ID ${req.params.medicalContactId} does not exist.`
+      });
+      return;
+    }
+
+  next();
+  
+};
+
+/**
+ * Checks if a medical contact with medicalContactId in req.params is active
+ */
+ const isMedicalContactActive = async (req: Request, res: Response, next: NextFunction) => {
+  const validFormat = Types.ObjectId.isValid(req.params.medicalContactId);
+  const medicalContact = validFormat ? await MedicalContactCollection.findOne(req.params.medicalContactId) : '';
+  if (!medicalContact.active) {
+    res.status(409).json({
+      error: `Medical contact with medical contact ID ${req.params.medicalContactId} is already deactivated.`
+    });
+    return;
+  }
+
+next();
+
+};
+
+/**
+ * Checks if a first name in req.body is valid, that is, it matches the name regex
+ */
+const isValidFirstName = (req: Request, res: Response, next: NextFunction) => {
+    const nameRegex = /^\w+$/i;
```

Comment:
I think this might be too limiting for users with non-english names. Maybe we could use `/^[a-z ,.'-]+$/i` instead? (same comment on last name)

Reply:
I think even that may be too restrictive (it could have issues with special character names). I think I'll switch to using ^(?!\s*$).+ instead (just checks for a non-empty string)